app/services/teacher_training_public_api/sync_all_providers_and_courses.rb

```diff
@@ -19,6 +19,16 @@ def self.call(recruitment_cycle_year: ::RecruitmentCycle.current_year, increment
       end
 
       TeacherTrainingPublicAPI::SyncCheck.set_last_sync(Time.zone.now)
+
+      if !incremental_sync && RequestStore.store[:full_sync]
+        data_store ||= RedisStore.new(key: RequestStore.store[:full_sync])
+        data = data_store.read ? JSON.parse(data_store.read) : {}
+        data.map { |_, value| value }
+
+        RequestStore.store[:full_sync] = false
```

Comment:
The current TTAPI runs as follows:
We kick off a `SyncAllProvidersAndCoursesWorker` job which then calls the teacher training API, upserts any providers in the same job but then schedules a `TeacherTrainingPublicAPI::SyncCourses` job for each provider at a set delay. When the `TeacherTrainingPublicAPI::SyncCourses` runs each job itself will also schedule a `TeacherTrainingPublicAPI::SyncSites` job as well.

When the `SyncAllProvidersAndCoursesWorker` will have completed it will have only attempted to sync providers. This means if we set this value to false here and in `SyncAllProvidersAndCoursesWorker` we won't have fully run a full sync, and will only capture errors for the provider updates. 

I was revisiting the initial approach which we were trying to avoid, and having a think around passing in `incremental: false` to the `TeacherTrainingPublicAPI::SyncCourses` and also `TeacherTrainingPublicAPI::SyncSites` job as well, where we can use this as a flag to raise an error if `course.changed?` etc. it does feel like the least complex approach after all and I think might solve most our problems especially with distinguishing between incremental and full syncs. Let me know your thoughts

Reply:
That sounds good. I will amend my PR to incorporate this changes.